core/types/types.cc

```diff
@@ -77,15 +77,27 @@ TypePtr Types::Float() {
 }
 
 TypePtr Types::arrayOfUntyped(sorbet::core::SymbolRef blame) {
-    static vector<TypePtr> targs{Types::untyped(blame)};
-    static auto res = make_type<AppliedType>(Symbols::Array(), move(targs));
-    return res;
+    if constexpr (!sorbet::track_untyped_blame_mode && !sorbet::debug_mode) {
```

Comment:
If I'm reading this correctly, this means that all debug mode builds will used uncached types for blamed untyped calls, whereas I would expect we would only want the no-caching behavior in `track_untyped_blame_mode`?  I suppose the rationale here is that you want to be able to track blames in the debugger if that would come up, and it's inconvenient to build an entirely separate blame-untyped build for such purposes.

Reply:
Yeah it's a holdover from how things used to work--this is the condition we use for all our blame tracking now.

Before we added a custom bazel config for track_untyped_blame_mode, the only flag we had was debug_mode. We added track_untyped_blame_mode so that you could track blames without also having to turn on all the ENFORCEs, so it would run fast on Stripe's codebase.

We thought about whether we should flip things over exclusively to branching on track_untyped_blame_mode, but we decided it would be nice to get some semblance of "it doesn't crash" coverage by having the typed blame tracking also run in CI.